File: mobile/androidApp/src/main/kotlin/com/sect/mobile/android/ui/components/SectDesignSystem.kt
// SECT Mobile — Design System Components (Savane EdTech)
// SECT-MOBILE-NAV-PHASE-E : composants DS unifiés Android
//
// Miroir des composants web /frontend/src/components/ds/ :
// - GlassCard (glassmorphism léger)
// - KenteDivider (motif kente tricolore lime/terre/or)
// - SectStatCard (carte métrique KPI)
// - SectProgressBar (barre animée)
package com.sect.mobile.android.ui.components

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.blur
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.sect.mobile.android.theme.*

// ════════════════════════════════════════════════════════
// GlassCard — glassmorphism léger (backdrop blur + transparence)
// ════════════════════════════════════════════════════════

/**
 * Carte avec effet glassmorphism léger (inspiré du web GlassModal).
 * Fond semi-transparent + bordure subtile + léger blur.
 *
 * @param modifier Modifier personnalisé
 * @param content Contenu de la carte
 */
@Composable
fun GlassCard(
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit
) {
    Card(
        modifier = modifier
            .clip(RoundedCornerShape(16.dp))
            .border(
                width = 1.dp,
                color = Color.White.copy(alpha = 0.3f),
                shape = RoundedCornerShape(16.dp)
            ),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surface.copy(alpha = 0.85f)
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 0.dp),
        shape = RoundedCornerShape(16.dp)
    ) {
        Column(content = content)
    }
}

// ════════════════════════════════════════════════════════
// KenteDivider — motif kente tricolore (lime/terre/or)
// ════════════════════════════════════════════════════════

/**
 * Séparateur avec motif kente — 3 bandes alternées vert lime / terre cuite / or.
 * Signature visuelle africaine subtile (inspiré du PDF epreuve-pdf-react.tsx).
 *
 * @param modifier Modifier personnalisé
 * @param thickness Hauteur du divider (défaut 4dp)
 */
@Composable
fun KenteDivider(
    modifier: Modifier = Modifier,
    thickness: Int = 4
) {
    Row(modifier = modifier.height(thickness.dp)) {
        // Bandes alternées : lime / terre / or / lime / terre / or
        val bands = listOf(SectLime, SectTerreCuite, SectGold)
        repeat(6) { i ->
            Box(
                modifier = Modifier
                    .weight(1f)
                    .background(bands[i % 3])
            )
        }
    }
}

// ════════════════════════════════════════════════════════
// SectStatCard — carte métrique KPI (miroir web StatCard)
// ════════════════════════════════════════════════════════

/**
 * Carte statistique avec valeur + label + icône + couleur accent.
 *
 * @param value Valeur affichée (ex: "15", "75%")
 * @param label Libellé (ex: "Épreuves", "Moyenne")
 * @param icon Icône Material (vector)
 * @param accentColor Couleur d'accent (défaut: vert lime)
 */
@Composable
fun SectStatCard(
    value: String,
    label: String,
    icon: @Composable () -> Unit,
    accentColor: Color = SectLime,
    modifier: Modifier = Modifier
) {
    GlassCard(modifier = modifier) {
        Row(
            modifier = Modifier.padding(12.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Box(
                modifier = Modifier
                    .size(40.dp)
                    .clip(RoundedCornerShape(10.dp))
                    .background(accentColor.copy(alpha = 0.15f)),
                contentAlignment = Alignment.Center
            ) {
                icon()
            }
            Spacer(modifier = Modifier.width(10.dp))
            Column {
                Text(
                    value,
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurface
                )
                Text(
                    label,
                    style = MaterialTheme.typography.labelSmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

// ════════════════════════════════════════════════════════
// SectProgressBar — barre de progression animée
// ════════════════════════════════════════════════════════

/**
 * Barre de progression animée avec gradient.
 *
 * @param progress Valeur 0..1
 * @param color Couleur de la barre (défaut: vert lime)
 * @param modifier Modifier personnalisé
 */
@Composable
fun SectProgressBar(
    progress: Float,
    modifier: Modifier = Modifier,
    color: Color = SectLime
) {
    val animatedProgress by animateFloatAsState(
        targetValue = progress.coerceIn(0f, 1f),
        animationSpec = tween(durationMillis = 600),
        label = "progress"
    )
    Box(
        modifier = modifier
            .height(8.dp)
            .clip(RoundedCornerShape(4.dp))
            .background(MaterialTheme.colorScheme.surfaceVariant)
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth(animatedProgress)
                .height(8.dp)
                .clip(RoundedCornerShape(4.dp))
                .background(
                    Brush.horizontalGradient(
                        listOf(color, color.copy(alpha = 0.7f))
                    )
                )
        )
    }
}

// ════════════════════════════════════════════════════════
// SectBadge — badge avec couleur sémantique
// ════════════════════════════════════════════════════════

/**
 * Badge compact avec couleur d'accent.
 *
 * @param text Texte du badge
 * @param color Couleur d'accent (défaut: vert lime)
 */
@Composable
fun SectBadge(
    text: String,
    color: Color = SectLime,
    modifier: Modifier = Modifier
) {
    Text(
        text = text,
        style = MaterialTheme.typography.labelSmall,
        fontWeight = FontWeight.Bold,
        color = color,
        modifier = modifier
            .clip(RoundedCornerShape(6.dp))
            .background(color.copy(alpha = 0.15f))
            .padding(horizontal = 8.dp, vertical = 4.dp)
    )
}

File: mobile/iosApp/iosApp/Utilities/SectDesignSystem.swift
//
//  SectDesignSystem.swift
//  SECT
//
//  SECT-MOBILE-NAV-PHASE-E : composants DS unifiés iOS (Savane EdTech)
//  Miroir de /frontend/src/components/ds/ + android SectDesignSystem.kt
//
import SwiftUI
import Shared

// ════════════════════════════════════════════════════════
// GlassCard — glassmorphism léger
// ════════════════════════════════════════════════════════

struct GlassCard<Content: View>: View {
    let content: () -> Content

    init(@ViewBuilder content: @escaping () -> Content) {
        self.content = content
    }

    var body: some View {
        content()
            .padding()
            .background(Color.savaneCard.opacity(0.85))
            .background(.ultraThinMaterial)
            .overlay(
                RoundedRectangle(cornerRadius: 16)
                    .stroke(Color.white.opacity(0.3), lineWidth: 1)
            )
            .cornerRadius(16)
            .shadow(color: Color.black.opacity(0.05), radius: 8, x: 0, y: 2)
    }
}

// ════════════════════════════════════════════════════════
// KenteDivider — motif kente tricolore (lime/terre/or)
// ════════════════════════════════════════════════════════

struct KenteDivider: View {
    var height: CGFloat = 4

    private let bands: [Color] = [.sectLime, .sectTerreCuite, .sectGold]

    var body: some View {
        HStack(spacing: 0) {
            ForEach(0..<6, id: \.self) { i in
                Rectangle()
                    .fill(bands[i % 3])
                    .frame(height: height)
            }
        }
    }
}

// ════════════════════════════════════════════════════════
// SectStatCard — carte métrique KPI
// ════════════════════════════════════════════════════════

struct SectStatCard: View {
    let value: String
    let label: String
    let icon: String
    let accentColor: Color

    init(value: String, label: String, icon: String, accentColor: Color = .sectLime) {
        self.value = value
        self.label = label
        self.icon = icon
        self.accentColor = accentColor
    }

    var body: some View {
        HStack(spacing: 10) {
            // Icône dans un cercle coloré
            Image(systemName: icon)
                .font(.title3)
                .foregroundColor(accentColor)
                .frame(width: 40, height: 40)
                .background(accentColor.opacity(0.15))
                .cornerRadius(10)

            VStack(alignment: .leading, spacing: 2) {
                Text(value)
                    .font(.title3).fontWeight(.bold)
                Text(label)
                    .font(.caption).foregroundColor(.secondary)
            }
            Spacer()
        }
        .padding(12)
        .background(Color.savaneCard.opacity(0.85))
        .background(.ultraThinMaterial)
        .overlay(
            RoundedRectangle(cornerRadius: 12)
                .stroke(Color.white.opacity(0.3), lineWidth: 1)
        )
        .cornerRadius(12)
    }
}

// ════════════════════════════════════════════════════════
// SectProgressBar — barre animée avec gradient
// ════════════════════════════════════════════════════════

struct SectProgressBar: View {
    let progress: Double
    var color: Color = .sectLime

    var body: some View {
        GeometryReader { geo in
            ZStack(alignment: .leading) {
                // Fond
                RoundedRectangle(cornerRadius: 4)
                    .fill(Color.savaneBorder)
                    .frame(height: 8)
                // Remplissage animé
                RoundedRectangle(cornerRadius: 4)
                    .fill(
                        LinearGradient(
                            colors: [color, color.opacity(0.7)],
                            startPoint: .leading,
                            endPoint: .trailing
                        )
                    )
                    .frame(width: geo.size.width * CGFloat(progress), height: 8)
                    .animation(.easeInOut(duration: 0.6), value: progress)
            }
        }
        .frame(height: 8)
    }
}

// ════════════════════════════════════════════════════════
// SectBadge — badge compact
// ════════════════════════════════════════════════════════

struct SectBadge: View {
    let text: String
    var color: Color = .sectLime

    var body: some View {
        Text(text)
            .font(.caption).fontWeight(.bold)
            .foregroundColor(color)
            .padding(.horizontal, 8)
            .padding(.vertical, 4)
            .background(color.opacity(0.15))
            .cornerRadius(6)
    }
}
